src/Cardano/Shell/Update/Lib.hs

```diff
@@ -0,0 +1,145 @@
+{-| Update module
+-}
+
+{-# LANGUAGE CPP                 #-}
+{-# LANGUAGE LambdaCase          #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+
+module Cardano.Shell.Update.Lib
+    ( UpdaterData(..)
+    , UpdateError(..)
+    , RunCmdFunc
+    , updaterData
+    , runUpdater
+    , runUpdater'
+    ) where
+
+import           Cardano.Prelude
+
+import qualified Data.Text as T
+import           Distribution.System (OS (..), buildOS)
+import           Prelude (String)
+import           System.Directory (doesFileExist, removeFile)
+import           System.Environment (getExecutablePath)
+import           System.Process (proc, waitForProcess, withCreateProcess)
+#ifdef mingw32_HOST_OS
+import           System.Win32.Process (getCurrentProcessId)
+#endif
+
+-- | Updater path, args, windows runner path, archive path
+data UpdaterData = UpdaterData
+    { udPath        :: !FilePath
+    , udArgs        :: ![Text]
+    , udWindowsPath :: Maybe FilePath
+    , udArchivePath :: Maybe FilePath
```

Comment:
You don't need this since you know whether it's a Windows platform or not.
You just need one of these.